tests/test_resolutions.py

```diff
@@ -9,19 +39,20 @@ def test_can_be_instantiated(self):
 
         assert isinstance(ic, IsClickable)
 
-    def test_matches_a_clickable_element(self):
-        element = mock.Mock()
+    @mock.patch("selenium.webdriver.remote.webelement.WebElement", spec=WebElement)
+    def test_matches_a_clickable_element(self, element):
         element.is_enabled.return_value = True
         element.is_displayed.return_value = True
         ic = IsClickable()
 
         assert ic._matches(element)
 
-    def test_does_not_match_unclickable_element(self):
-        invisible_element = mock.Mock()
+
+    @mock.patch("selenium.webdriver.remote.webelement.WebElement", spec=WebElement)
+    @mock.patch("selenium.webdriver.remote.webelement.WebElement", spec=WebElement)
```

Comment:
I'm not sure i understand the difference here. Why pass it in like this instead of creating the mock with a spec in the test?

For completeness' sake, i agree that we may as well spec against the WebElement, but i don't think we need to do a patch.

Reply:
Patching felt safer to me so that all of the various attributes and methods of WebElement are included in the mock.  I know for my own sake, I was worried I would miss something vital in the mock up that would prove we're comparing apples to apples.  It might be a bit extra but I didn't feel like it hurt anything.

Reply:
Doing the `patch` just adds a `mock.Mock` with the spec you specified, but replacing the object at the path you gave with that mock. So in this case it's essentialy identical to just doing `mock.Mock(spec=WebElement)` since we don't interact with anything that actually creates a `WebElement`.

Reply:
I don't have strong feelings about using the decorator vs creating the mock in the method.  I think I am just used to patching objects via decorator. 
It somehow feels... sophisticated?  Refined?  upper-crust even?  🤣